lib/semian/redis.rb

```diff
@@ -0,0 +1,58 @@
+require 'semian'
+require 'semian/adapter'
+require 'redis'
+
+class Redis
+  class SemianError < Redis::BaseConnectionError
+    include ::Semian::AdapterError
+  end
+
+  ResourceOccupiedError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module Redis
+    include Adapter
+
+    ResourceOccupiedError = ::Redis::ResourceOccupiedError
+    CircuitOpenError = ::Redis::CircuitOpenError
+
+    # The naked methods are exposed as `raw_query` and `raw_connect` for instrumentation purpose
```

Comment:
I asked for this comment for the MySQL adapter. I think this comment is very helpful.